modules/sql/managed-instance/vulnerability-assessment/main.bicep

```diff
@@ -39,12 +44,21 @@ resource managedInstance 'Microsoft.Sql/managedInstances@2022-02-01-preview' exi
   name: managedInstanceName
 }
 
+module roleAssignment '.bicep/nested_storageRoleAssignment.bicep' = if (!(useStorageAccountAccessKey)) {
+  name: guid(managedInstance.id, managedInstance.name, 'Storage Blob Data Contributor')
+  scope: resourceGroup(splitStorageAccountResourceId[4])
+  params: {
+    storageAccountName: last(splitStorageAccountResourceId)
+    managedInstancePrincipalId: managedInstance.identity.principalId
```

Comment:
Just curious, is there a general requirement that it must be a system-assigned identity? And if it can also be a User-Assigned Identity, should we support it? Also, does this break omif one did not active the system assigned identity as it would then try and access a property which isn't available (if I'm not mistaken)

Reply:
This isn't easy to solve 😅

Reply:
We should also probably think about adding a test case for this.

Reply:
Scenarios potentially exist for RBAC:
1. SQL MI uses System Identity, and the existing Storage account is in the same RG.
2. SQL MI uses System Identity, and the existing Storage account is in a different RG.
3. SQL MI uses User Identity, and the existing Storage account is in the same RG.
4. SQL MI uses User Identity, and the existing Storage account is in a different RG.

3/4: I think it's better 'Not' to perform a Role assignment as this can be treated as a pre-req since the User identity would also exist before the SQL MI. 

1/2: I think we can optionally support this scenario, by an explicit parameter (i.e. `createStorageRoleAssignment`) to perform RBAC instead of doing it automatically if `useStorageAccountAccessKey` is empty. If it's automatic, we are assuming the user/pipeline creating SQL MI has `owner` rights on either the same RG or the different RG containing the storage, which could also assume its Sub level `owner` role as well thus failing the deployment for end users if they don't have the permissions. 

Thoughts?

Reply:
According to the documentation, user assigned managed identities  are not supported, so the requirement would be to use system assigned. https://learn.microsoft.com/en-us/azure/azure-sql/database/sql-database-vulnerability-assessment-storage?view=azuresql#prerequisites, so we'd be looking at just scenarios 1/2. 

Regarding RBAC, it's indeed a valid point that if it's automatic RBAC permissions would be required. Initially I was thinking  we make this assumption for all modules, however in this case since it would be automatic, the user is unable to choose not to use the functionality which could fail the deployment. If we are going to support this scenario, I agree that we'd probably need a parameter to perform RBAC as you suggested.

Reply:
Met with @AlexanderSehr and we agreed it makes sense to use the createStorageRoleAssignment param to indicate whether to create the role assignment. Even though this would fail if the role assignment already exists on the storage account, performing the role assignment more closely matches the behavior in the portal. I added a note on the parameter to caution on this.